contracts/child/ChildToken/DappTokens/UChildDAI.sol

```diff
@@ -0,0 +1,53 @@
+pragma solidity 0.6.6;
+
+import {UChildERC20} from "../UpgradeableChildERC20/UChildERC20.sol";
+
+contract UChildDAI is UChildERC20 {
+    // bytes32 public constant PERMIT_TYPEHASH = keccak256("Permit(address holder,address spender,uint256 nonce,uint256 expiry,bool allowed)");
+    bytes32 public constant PERMIT_TYPEHASH = 0xea2aa0a1be11a07ed86d755c93467f4f82362b452371d1ba94d1715123511acb;
+
+    // --- Alias ---
+    function push(address usr, uint wad) external {
+        transferFrom(msg.sender, usr, wad);
+    }
+    function pull(address usr, uint wad) external {
+        transferFrom(usr, msg.sender, wad);
+    }
+    function move(address src, address dst, uint wad) external {
+        transferFrom(src, dst, wad);
+    }
+
+    // --- Approve by signature ---
+    function permit(
+        address holder,
+        address spender,
+        uint256 nonce,
+        uint256 expiry,
+        bool allowed,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external {
```

Comment:
Should we add `require(msg.sender != address(this), "Cannot allow meta tx with permit")` ?

Reply:
Since the permit doesn't use `msg.sender`, it doesn't matter, right? But `nonces[address(this)]` will increase.

Reply:
Added check